packages/contracts/contracts/Artonomous.sol

```diff
@@ -30,19 +35,50 @@ contract Artonomous {
         pieceToken.mint(this, block.number, currentGeneratorHash);
         currentAuction = Auction({
             blockNumber: block.number,
-            endTime: now + AUCTION_LENGTH
+            endTime: now + AUCTION_LENGTH,
+            startingPrice: getStartingPrice()
         });
         emit ArtonomousAuctionStarted(block.number, currentGeneratorHash);
     }
 
+    // placeholder
+    function getStartingPrice() internal pure returns (uint) {
+        return 1000000000000;
+    }
+
+    function buyArt() external payable {
+        uint blockNumber = currentAuction.blockNumber;
+        require(blockNumber > 0);
```

Comment:
I'm not 100% sure this is necessary? Trying to understand this.

Also: if we are going to use blockhashes, then does this check still matter?

Reply:
probably not necessary anymore. It was back when users had to manually start auctions since this check just makes sure an auction is in progress, but now we always have an auction in progress. I'll set up a PR removing it soon.

I was thinking it made more sense to just keep the blockNumber on the smart contract and just get the blockHash on the client from the blockNumber. If the blockNumber is what identifies an auction we can just use that as the ID of the minted token as well.